### PR TITLE
Replace XKit.Console calls with console.log. Fixes #483

### LINKED ISSUE #483: Deprecate XKit.console
`XKit.console` is a mess and really isn't all that useful to us for debugging.  It should be deprecated in favour of native `console.log` where absolutely necessary.

### PR DESCRIPTION
**Pull request by nepeat**
it will never be missed

### CODE REVIEW

File: Extensions/blacklist.js [hunk 1]
```diff
@@ -566,7 +566,7 @@ XKit.extensions.blacklist = new Object({
 
 			} catch(e) {
 
-				// XKit.console.add("Can't parse post: " + e.message);
+				// console.log("Can't parse post: " + e.message);
```
Comment (0xazure) [inline Extensions/blacklist.js]: This is commented out, so should be removed entirely.

File: Extensions/notifications_plus.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 //* TITLE Notifications+ **//
-//* VERSION 1.5 REV C **//
+//* VERSION 1.5.1 **//
```
Comment (0xazure) [inline Extensions/notifications_plus.js]: Version number was incremented incorrectly.  `REV C` + 1 would be `REV D`, which maps to `1.5.4`.

File: Extensions/old_notifications.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 //* TITLE Classic Notifications **//
-//* VERSION 0.6 REV B **//
+//* VERSION 0.6.1 **//
```
Comment (0xazure) [inline Extensions/old_notifications.js]: Version number was incremented incorrectly. `REV B` + 1 would be `REV C`, which maps to `0.6.3`.

File: Extensions/xkit_patches.js [hunk 4]
```diff
@@ -1854,7 +1853,7 @@ XKit.tools.dump_config = function(){
 				XKit.page.standard = true;
 				XKit.init_extension();
 			} else {
-				// XKit.console.add("In IFRAME, location: " + document.location.href);
+				// console.log("In IFRAME, location: " + document.location.href);
```
Comment (0xazure) [inline Extensions/xkit_patches.js]: This has been commented out, so should be removed entirely.

File: Extensions/xkit_patches.js [hunk 5]
```diff
@@ -1952,7 +1951,7 @@ XKit.tools.dump_config = function(){
 
 				$("#xkit-notifications").append(m_html);
 
-				// XKit.console.add(" Notification > " + message);
+				// console.log(" Notification > " + message);
```
Comment (0xazure) [inline Extensions/xkit_patches.js]: This has been commented out, so should be removed entirely.

File: Extensions/xwidgets.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 //* TITLE XWidgets **//
-//* VERSION 0.3 REV A **//
+//* VERSION 0.3.1 **//
```
Comment (0xazure) [inline Extensions/xwidgets.js]: Version number was incremented incorrectly. `REV A` + 1 would be `REV B`, which maps to `0.3.2`.

File: xkit.js [hunk 7]
```diff
@@ -623,53 +617,8 @@ XKit = {
...
 		add: function(text) {
-			if ($("#xkit_console").length > 0) {
-				$("#xkit_console").append("<br/>" + text);
-				var objDiv = document.getElementById("xkit_console");
-				objDiv.scrollTop = objDiv.scrollHeight;
-			} else {
-				XKit.console.cache = XKit.console.cache + ("<br/>" + text);
-			}
-			console.log(text);
+			console.warn("[DEPRECATED] XKit.console", text);
```
Comment (0xazure) [inline xkit.js]: Methods other than `log()` on console [are non-standard](https://developer.mozilla.org/en/docs/Web/API/console) and should not be used.

File: Extensions/servant.js [hunk 8]
```diff
@@ -1,5 +1,5 @@
 //* TITLE Servant **//
-//* VERSION 0.5.9 **//
+//* VERSION 0.6.0 **//
```
Comment (0xazure) [inline Extensions/servant.js]: Version number incorrectly incremented.  Should be +1 to `PATCH`, which would be `0.5.10`.

**Comment by Wolvan** [comment]
From Gitter
@nepeat you made a bit of a mistake with the versions
REV A = .0
REV B = .1
REV C = .2 and so on
xkit_updates parses that way, thus turning REV C into .1 would be a version downgrade and would not update the extension
also, 0.5.9 doesn't get bumped to 0.6.0, it gets bumped to 0.5.10

**Comment by 0xazure** [comment]
There' a bunch of whitespace noise (see the diffs for `classic_tags`, `mass_plus`) that should be reverted to reduce code churn.

Also, these changes will make the browser's console very noisy.  The idea behind #483 was to _remove_ logging entirely unless absolutely necessary (ie. when XKit encounters a state that it cannot recover from).  `XKit.console` should not just be replaced with `console.log()`, but rather removed completely.  There are better debugging tools than raw `console.log()` statements.

**Comment by Wolvan** [comment]
@0xazure it isn't going to generate any more noise than it currently does. It also gives us a far better way to debug now, since we have the proper stacktraces and line locations. We can remove the unneeded console.log statements later in my opinion, for now having XKit.console gone is a good start

**Comment by 0xazure** [comment]
True, I'd forgotten that the XKit console delegates down to the browser console in addition to its own output.  However,

> There are better debugging tools than raw `console.log()` statements.

My point still stands, and I've already have this conversation with @nightpool.  If we're going to remove `XKit.console`, we should be _removing_ the useless debugging statements at the same time, not kicking the change further down the road.

**Comment by Wolvan** [comment]
We do not have these better logging tools so far though, so we would have to wait until our next core release. Or do you see another way of loading them without core?

**Comment by 0xazure** [comment]
@Wolvan I didn't say anything about _logging_ tools, I said _debugging_ tools.  The browser console has good debugging support built-in, adding extra logging or modules isn't going to help us much there.

Additionally, the logging statements as they are are pretty useless.  They don't _tell_ us much that's actionable.  I would wager 90% of the existing `XKit.console` statements are leftover development debugging logs that should never have made it into production to begin with.

**Comment by Wolvan** [comment]
@0xazure sorry, I misread that one. I thought you were talking about that one logging tool we found a few weeks ago which allows us to set log levels and stuff. The thing is, the log statements have been useless to us because they did not print to any stacktrace afaik

**Comment by nightpool** [comment]
>If we're going to remove XKit.console, we should be removing the useless debugging statements, not kicking the change further down the road.

at least remove the event handler/fan mail ones....

**Comment by 0xazure** [comment]
@Wolvan see [the Gitter chat from Oct 6th](https://gitter.im/new-xkit/XKit?at=5614715bfd5c74d7141433ce).

> The thing is, the log statements have been useless to us because they did not print to any stacktrace afaik

But regular `console.log()` doesn't give us a stacktrace anyways?  We'd only get a stacktrace if there's an error or warning being thrown.

**Comment by Wolvan** [comment]
Well, stacktrace might be the wrong name, I meant the line number and the script where it happened

**Comment by 0xazure** [comment]
@Wolvan but that was fixed in #299?

**Comment by finagle29** [comment]
#299 tells us that XKit.console.add was called in xkit.js, not where the XKit.console.add call originated from

**Comment by Wolvan** [comment]
@0xazure no, every XKit.console.add call always just shows the line number of the definition of XKit.console.add. #299 just made it so that each XKit script gets its own JS file in the debugger

**Comment by 0xazure** [comment]
Oh, then I misunderstood the purpose of the changes in #299 .

The point about line numbers is moot anyways because the log statements are unique enough that they're easily searchable.

File: xkit.js [hunk 9]
```diff
@@ -231,12 +225,12 @@ XKit = {
 		max_try_count: 5,
 		// TODO: implement as module, lose most of this code
 		github_fetch: function(path, callback, fallback) {
-			var url = 'https://new-xkit.github.io/XKit/Extensions/dist/' + path;
+			var url = 'https://localhost:31337/Extensions/dist/' + path;
```
Comment (bvtsang) [inline xkit.js]: Please revert this line.

**Comment by nepeat** [comment]
Accidental localhost commit has been reverted.

**Comment by nepeat** [comment]
Closed because oh god, I broke git.

**Comment by nightpool** [comment]
rebase and force push

**Pull request closed without merging**